File: benchmark/nodejs/power.js
function power(base, exp) {
    let result = 1n;
    const b = BigInt(base);
    for (let i = 0; i < exp; i = i + 1) {
        result = result * b;
    }
    return result;
}

let sum = 0n;
const start = Date.now();
for (let i = 0; i < 10000; i = i + 1) {
    sum = sum + power(2, 1000);
}
const end = Date.now();
console.log(`Sum: ${sum}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/python/power.py
import time

def power(base, exp):
    result = 1
    for i in range(exp):
        result = result * base
    return result

sum = 0
start = time.time() * 1000
for i in range(10000):
    sum = sum + power(2, 1000)
end = time.time() * 1000
print(f"Sum: {sum}")
print(f"Time: {end - start:.0f}ms")
